### PR TITLE
csi-driver: make linux iscsi/disk/multipath tuning configurable

### PR DESCRIPTION
**Pull request by shivamerla**
* Problem:
  * currently our drivers are applying changes from fixed template and not user
  * configurable. settings like queue_depth, max_sectors_kb should be configurable
  * based on workloads.
* Implementation:
  * Added config-map called hpe-linux-config with all template settings we have
  * User can edit as required and re-deploy node plugin to apply those changes
  * This will be first stage of changes for csi1.0 and later can be modified to control
  * 3par multipath settings etc.
* Testing: deployed using helm/yaml and verified settings are applied.
* Review: gcostea, rkumar, sbyadarahalli
* Bug: https://nimblejira.nimblestorage.com/browse/NLT-
Signed-off-by: Shiva Krishna, Merla <shivakrishna.merla@hpe.com>

### CODE REVIEW

**Comment by shivamerla** [comment]
Testing:

Deploy the plugin using helm charts and verify queue_depth of 256 configured:

```root@node-u1804k8s-cl2-1:/home/co-deployments/helm/charts# cat /etc/iscsi/iscsid.conf | grep queue_depth
# To control the device's queue depth set node.session.queue_depth
node.session.queue_depth = 256```

Edit configmap to change queue_depth to 512

```
root@node-u1804k8s-cl2-1:/home/co-deployments/helm/charts# kubectl edit configmap hpe-linux-config
```

Redeploy node plugin:

```
root@node-u1804k8s-cl2-1:/home/co-deployments/helm/charts# kubectl delete pod -l app=hpe-csi-node
pod "hpe-csi-node-qr82z" deleted
pod "hpe-csi-node-w5l7g" deleted


root@node-u1804k8s-cl2-1:/home/co-deployments/helm/charts# cat /etc/iscsi/iscsid.conf | grep queue_depth
# To control the device's queue depth set node.session.queue_depth
node.session.queue_depth = 512```

**Comment by shivamerla** [comment]
> Will we overwrite with the shorter file ?
Yes, currently we are not setting disk recommendations with our drivers, but they were there in config.json. so just removed them. I was thinking of adding disk tuning, but that needs more validation with udev rules etc. So we can handle that with 1.1

File: helm/charts/hpe-csi-driver/README.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ The following table lists the configurable parameters of the HPE-CSI chart and t
 
 It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-deployments/blob/master/helm/charts/hpe-csi-driver/values.yaml) file and edit it to fit the environment the chart is being deployed to. Download and edit the sample file.
 
+**Note:** HPE CSI Driver for Kubernetes automatically configures Linux iSCSI/Disk/Multipath settings based from [config.json](https://raw.githubusercontent.com/hpe-storage/co-deployments/master/helm/charts/hpe-csi-driver/files/config.json). In order to tune these values, edit the config map with `kubectl edit configmap hpe-linux-config -n kube-system` and restart node plugin using `kubectl delete pod -l app=hpe-csi-node` to apply.
```
Comment (rgcostea) [inline helm/charts/hpe-csi-driver/README.md]: nit: based on

File: operators/hpe-csi-operator/README.md [hunk 2]
```diff
@@ -31,6 +31,8 @@ Parameter list:<br/>
 4. ``values.yaml`` is the customized helm-chart configuration parameters. This is a **required parameter** and must contain a valid backend HPE storage system. All parameters that need a non-default value must be specified in this file.
 Refer to [Configuration for values.yaml.](https://github.com/hpe-storage/co-deployments/tree/master/helm/charts/hpe-csi-driver#configuration--installation) for details about various parameters.
 
+**Note:** HPE CSI Driver for Kubernetes automatically configures Linux iSCSI/Disk/Multipath settings based from [config.json](https://raw.githubusercontent.com/hpe-storage/co-deployments/master/helm/charts/hpe-csi-driver/files/config.json). In order to tune these values, edit the config map with `kubectl edit configmap hpe-linux-config -n hpe-csi` and restart node plugin using `kubectl delete pod -l app=hpe-csi-node` to apply.
```
Comment (rgcostea) [inline operators/hpe-csi-operator/README.md]: nit: based on

File: helm/charts/hpe-csi-driver/README.md (hunk 1 above)
Comment (shivamerla) [reply helm/charts/hpe-csi-driver/README.md]: done..

**Pull request merged**